### ISSUE TITLE
Reoccurring Pytest Failure on Ubuntu Python 3.8/3.9

### ISSUE THREAD

**Issue by bpaul4** [IDAES/idaes-pse#644]
Recently the test called by idaes/surrogate/pysmo/tests/test_utils.py::TestNumpyEvaluator::test_eval_numpy has been failing due to a seemingly minor precision error (see GitHub Actions pytest output below). The fix may be simply updating the called Numpy version, or related to Ubuntu-specific processes as this error does not occur for Windows either locally or through GitHub Actions.

Ubuntu w/ Python 3.8
```
______________________ TestNumpyEvaluator.test_eval_numpy ______________________

self = <idaes.surrogate.pysmo.tests.test_utils.TestNumpyEvaluator object at 0x7f335d3e28b0>

    @pytest.mark.unit
    def test_eval_numpy(self):
        m = ConcreteModel()
        m.p = Param([1,2], mutable=True)
        m.x = Var()
    
        data = np.array([[0,-1,2],
                         [.1,.2,.3],
                         [4,5,6]])
        cMap = ComponentMap()
        cMap[m.p[1]] = data[0]
        cMap[m.p[2]] = data[1]
        cMap[m.x] = data[2]
    
        npe = NumpyEvaluator(cMap)
    
        result = npe.walk_expression(sin(m.x))
/home/runner/work/idaes-pse/idaes-pse/idaes/apps/matopt/opt/pyomo_modeling.py:1355: SyntaxWarning: "is not" with a literal. Did you mean "!="?
>       assert result[0] == sin(4)
  if PosZic is not 0:
E       assert -0.7568024953079284 == -0.7568024953079282
/home/runner/work/idaes-pse/idaes-pse/idaes/apps/matopt/opt/pyomo_modeling.py:1357: SyntaxWarning: "is not" with a literal. Did you mean "!="?
E        +  where -0.7568024953079282 = sin(4)
  if NegZic is not 0:

/home/runner/work/idaes-pse/idaes-pse/idaes/apps/matopt/opt/pyomo_modeling.py:1471: SyntaxWarning: "is not" with a literal. Did you mean "!="?
/home/runner/work/idaes-pse/idaes-pse/idaes/surrogate/pysmo/tests/test_utils.py:55: AssertionError
  if PosZic is not 0:

/home/runner/work/idaes-pse/idaes-pse/idaes/apps/matopt/opt/pyomo_modeling.py:1473: SyntaxWarning: "is not" with a literal. Did you mean "!="?
---------- coverage: platform linux, python 3.8.12-final-0 -----------
  if NegZic is not 0:
Coverage XML written to file coverage.xml
```

Ubuntu w/ Python 3.9
```
______________________ TestNumpyEvaluator.test_eval_numpy ______________________

self = <idaes.surrogate.pysmo.tests.test_utils.TestNumpyEvaluator object at 0x7ff8e4ad3af0>

    @pytest.mark.unit
    def test_eval_numpy(self):
        m = ConcreteModel()
        m.p = Param([1,2], mutable=True)
        m.x = Var()
    
        data = np.array([[0,-1,2],
                         [.1,.2,.3],
                         [4,5,6]])
        cMap = ComponentMap()
        cMap[m.p[1]] = data[0]
        cMap[m.p[2]] = data[1]
        cMap[m.x] = data[2]
    
        npe = NumpyEvaluator(cMap)
    
        result = npe.walk_expression(sin(m.x))
>       assert result[0] == sin(4)
E       assert -0.7568024953079284 == -0.7568024953079282
E        +  where -0.7568024953079282 = sin(4)

/home/runner/work/idaes-pse/idaes-pse/idaes/surrogate/pysmo/tests/test_utils.py:55: AssertionError
```

**Comment by lbianchi-lbl** [comment]
Thanks for reporting this @bpaul4. As far as I can tell this looks like a floating point precision issue (the discrepancy is in the last digit). The more interesting question for me is what caused the test to start failing in the CI, since it doesn't seem to be tied to a change in the code itself. My guess would be something in numpy and/or the system libraries in the Ubuntu version that's being used for the CI (the OS version itself has not been changed, or at least not explicitly).

To fix this, unless this particular test requires extra details that I might be overlooking, using `pytest.approx()` with a suitable precision/tolerance should be enough (and, more in general, something we should probably encourage developers to use when comparing floating point numbers).

**Comment by ksbeattie** [comment]
Fixed by #647
